src/Database/Connection.php

```diff
@@ -694,6 +694,25 @@ public function log($sql)
     }
 
     /**
+     * {@inheritDoc}
+     */
+    public function supportsCrossWith(ConnectionInterface $target)
+    {
+        $sourceConfig = $this->config();
+        $targetConfig = $target->config();
+
+        if ($sourceConfig['driver'] !== $targetConfig['driver']) {
+            return false;
+        }
+
+        if ((isset($sourceConfig['host'])) && ($sourceConfig['host'] !== $targetConfig['host'])) {
```

Comment:
Don't forget ports. Someone will run 2 copies of mysql on different ports and complain that its 'broken'.